logic/transaction.js

```diff
@@ -15,6 +15,15 @@ var self, modules, __private = {}, genesisblock = null;
 
 __private.types = {};
 
+// Private methods
+__private.calculateFee = function (trs, sender) {
+	var fee = 0;
+	if (trs.data) {
+		fee = constants.fees.data;
+	}
+	return __private.types[trs.type].calculateFee.call(this, trs, sender) + fee || false;
```

Comment:
@SargeKhan In several places we already use safer `bignumber` approach, so for new code we should continue with that approach. Yep, will be nice to use that approach globally - I'll create new issue for that.